spec/rubocop/cop/performance/use_zip_to_wrap_array_contents_spec.rb

```diff
@@ -0,0 +1,288 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Performance::UseZipToWrapArrayContents, :config do
+  context 'when using map with array literal' do
+    it 'registers an offense and corrects to use zip with no arguments' do
+      expect_offense(<<~RUBY)
+        [1, 2, 3].map { |id| [id] }
+                  ^^^^^^^^^^^^^^^^^ Use `.zip` instead of `.map { |id| [id] }`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        [1, 2, 3].zip
+      RUBY
+    end
+  end
+
+  context 'when using map with a short iterator name' do
+    it 'registers an offense and corrects to use zip with no arguments' do
+      expect_offense(<<~RUBY)
+        [1, 2, 3].map { |e| [e] }
+                  ^^^^^^^^^^^^^^^ Use `.zip` instead of `.map { |id| [id] }`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        [1, 2, 3].zip
+      RUBY
+    end
+  end
+
+  context 'when using map on a range with another iterator name' do
+    it 'registers an offense and corrects' do
+      expect_offense(<<~RUBY)
+        (1..3).map { |x| [x] }
+               ^^^^^^^^^^^^^^^ Use `.zip` instead of `.map { |id| [id] }`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        (1..3).zip
+      RUBY
+    end
+  end
+
+  context 'when using map in a do end block' do
+    it 'registers an offense and corrects' do
+      expect_offense(<<~RUBY)
+        (a..b).map do |m| [m] end
+               ^^^^^^^^^^^^^^^^^^ Use `.zip` instead of `.map { |id| [id] }`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        (a..b).zip
+      RUBY
+    end
+  end
+
+  context 'when using map in a chain' do
+    it 'registers an offense and corrects' do
+      expect_offense(<<~RUBY)
+        [nil, tuple].flatten.map { |e| [e] }.call
+                             ^^^^^^^^^^^^^^^ Use `.zip` instead of `.map { |id| [id] }`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        [nil, tuple].flatten.zip.call
+      RUBY
+    end
+  end
+
+  context 'when the map block does not contain an array literal' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].map { |id| id }
+      RUBY
+    end
+  end
+
+  context 'when using collect' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].collect { |id| [id] }
+      RUBY
+    end
+  end
+
+  context 'when using select with an array literal' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1, 2, 3].select { |id| [id] }
+      RUBY
+    end
+  end
+
+  context 'when making an array of arrays using each_with_object' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        [1,2,3].each_with_object([]) {|id, object| object << [id]}
```

Comment:
It would be cool to add future, or current, support for well known other patterns like this

Reply:
Yes! Is this a change request? I added these just so we are exactly specifying the behavior. I do think it can be extended and made to work for this kind of example, as needed by the community.